.github/workflows/Intel_Parallelworks_CI.yaml

```diff
@@ -13,21 +13,29 @@ on:
   pull_request:
     branches:
       - main
+
+#this should cancel in progress ci runs for the same PR 
+#(e.g. a second commit on the same PR comes in while CI is still running)
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
```

Comment:
Do all of the actions below wait until confirmation of a cancellation?  I guess I'm concerned a cancellation is ignored and the updated actions continue concurrently in the separate SHA macro-based workspace.

Reply:
I could give this a test.  In theory, it is supposed to only allow one run in the concurrency group to go at a time and the most recent run will trigger a cancel of the older run.